util/generate-ebs-substack.py

```diff
@@ -0,0 +1,100 @@
+
+
+numberOfVol = 5
+
+
+t = Template()
+AvailabilityZone = t.add_parameter(Parameter("AvailabilityZone",
+                                             Type="String",
+                                             Description="Availability Zone the cluster will launch into. THIS IS REQUIRED"))
+VolumeSize = t.add_parameter(Parameter("VolumeSize",
+                                       Type="CommaDelimitedList",
+                                       Description = "Size of EBS volume in GB, if creating a new one",
+                                       Default = "20, 20, 20"))
+VolumeType = t.add_parameter(Parameter("VolumeType",
+                                       Type="CommaDelimitedList",
+                                       Description="Type of volume to create either new or from snapshot",
+                                       Default = "gp2, gp2, gp2"))
+VolumeIOPS = t.add_parameter(Parameter("VolumeIOPS",
+                                       Type="CommaDelimitedList",
+                                       Description= "Number of IOPS for volume type io1. Not used for other volume types.",
+                                       Default= "100, 100, 100"))
+EBSEncryption = t.add_parameter(Parameter("EBSEncryption",
+                                          Type="CommaDelimitedList",
+                                          Description="Boolean flag to use EBS encryption for /shared volume. (Not to be used for snapshots)",
+                                          Default= "false, false, false"))
+EBSKMSKeyId = t.add_parameter(Parameter("EBSKMSKeyId",
+                                        Type="CommaDelimitedList",
+                                        Description= "KMS ARN for customer created master key, will be used for EBS encryption",
+                                        Default="NONE, NONE, NONE"))
+EBSVolumeId = t.add_parameter(Parameter("EBSVolumeId",
+                                        Type="CommaDelimitedList",
+                                        Description= "Existing EBS volume Id",
+                                        Default= "NONE, NONE, NONE"))
+EBSSnapshotId = t.add_parameter(Parameter("EBSSnapshotId",
+                                          Type="CommaDelimitedList",
+                                          Description="Id of EBS snapshot if using snapshot as source for volume",
+                                          Default= "NONE, NONE, NONE"))
+EBSVolumeNum = t.add_parameter(Parameter("NumberOfEBSVol",
+                                         Type="Number",
+                                         Description= "Number of EBS Volumes the user requested, up to 10",
+                                         Default="1"))
+
+UseVol = [None]*numberOfVol
+UseExistingEBSVolume = [None]*numberOfVol
+v = [None]*numberOfVol
+
+for i in range(numberOfVol):
+    if i == 0:
+        CreateVol = t.add_condition("Vol" + str(i + 1) + "_CreateEBSVolume",
```

Comment:
use python string interpolation. It gets rid of need for `str()`. See https://www.programiz.com/python-programming/string-interpolation#%-formatting